### ISSUE TITLE
Unable to deobfuscate this JAR

### ISSUE THREAD

**Issue by wtf911** [java-deobfuscator/deobfuscator#311]
I'm unable to deobfuscate this, I just know it's using Stringer of some sort.

[applauncher.zip](https://github.com/java-deobfuscator/deobfuscator/files/2441625/applauncher.zip)

**Comment by ThisTestUser** [comment]
[applauncherDeob.zip](https://github.com/java-deobfuscator/deobfuscator/files/2444061/applauncherDeob.zip)
Use DuplicateRenamer if you want to rename the fields/methods.

**Comment by wtf911** [comment]
I'm not sure what I did wrong to not be able to get it deobfuscated but I appreciate you taking the time. If out of curiosity or an urge to be helpful perhaps you could go a step further with me. Looking inside the one you sent back to me I'm able to find that in the L class it links to: https://legitcheats.club/launcher/PRIVATE/app.jar

[app.zip](https://github.com/java-deobfuscator/deobfuscator/files/2444261/app.zip)

Would it be possible to deobfuscate that one as well? It seems to be using the same Stringer obfuscation.

Here's where my lack of Java knowledge comes in though and beyond deobfuscation perhaps you could offer some insight. This is for a game and when you open the first .jar you get:

 
![image](https://user-images.githubusercontent.com/5151430/46443638-628a5c00-c73c-11e8-8179-7a7a8860ac62.png)

Once you login you get:

![image](https://user-images.githubusercontent.com/5151430/46443674-92396400-c73c-11e8-97a4-88507b851e9a.png)

Being a bronze subscription level doesn't give access to the Aimbot v0.6 function. When you try to tick the checkbox for it you get this:

![image](https://user-images.githubusercontent.com/5151430/46443717-beed7b80-c73c-11e8-8ead-96c71771e852.png)

Can you see anyway to enable the aimbot by modifying it or through a workaround?

Thank you!

**Comment by ThisTestUser** [comment]
[appDeob.zip](https://github.com/java-deobfuscator/deobfuscator/files/2444594/appDeob.zip)
Since I'm unfamiliar with the program and I don't have time, I can't help you with modifying the file.
The deobfuscated file is above if you want to take a look. 
If you are going to modify the file, modify the original file (deobbed file will probably have verify errors) using JByteMod or a similar tool.

**Comment by wtf911** [comment]
Thank you for the help. After going over the deobfuscated files I think I may have found the easiest attack vector. There is a URL it goes to in order to grab the account subscription level.

`URL localURL = new URL("https://*****/CheatLauncher/Profiles/" + paramString + "/data.txt");`

That is in the applauncherDeob.zip you sent back to me at the beggining. The data.txt simply holds the subscription level for the user. If I can change that URL then I can host my own data.txt with whatever subscription level I want.

In other words I would like to change the:
`URL("https://*****/CheatLauncher/Profiles/" + paramString + "/data.txt");`
To instead be:
`URL("http://127.0.0.1/data.txt");`

I did get JByteMod as you suggested but I'm not sure how to modify the original applauncher.jar to achieve this. Would you be able to help with this or give me some more direction?

**Comment by wtf911** [comment]
I found another way to achieve the same thing. Instead of having the URL changed within the file I was able to use a proxy to reroute the request. I used Fiddler to get the HTTPS traffic and opened the .jar with java arguments to make it use the Fiddler proxy and certificate. Then I made a rule in Fiddler to redirect the URL to my own. Thank you again for the help.
